ComponentKit/Utilities/CKComponentGestureActions.h

```diff
@@ -17,44 +17,44 @@ typedef void (*CKComponentGestureRecognizerSetupFunction)(UIGestureRecognizer *)
 
 /**
  Returns a view attribute that creates and configures a tap gesture recognizer to send the given CKComponentAction.
-
+ 
  @param action Sent up the responder chain when a tap occurs. Sender is the component that created the view.
-        Context is the gesture recognizer. May be NULL, in which case no action will be sent.
+ Context is the gesture recognizer. May be NULL, in which case no action will be sent.
  */
-CKComponentViewAttributeValue CKComponentTapGestureAttribute(CKComponentAction action);
+CKComponentViewAttributeValue CKComponentTapGestureAttribute(const CKTypedComponentAction<UIGestureRecognizer *> &action);
 
 /**
  Returns a view attribute that creates and configures a pan gesture recognizer to send the given CKComponentAction.
-
+ 
  @param action Sent up the responder chain when a pan occurs. Sender is the component that created the view.
-        Context is the gesture recognizer. May be NULL, in which case no action will be sent.
+ Context is the gesture recognizer. May be NULL, in which case no action will be sent.
  */
-CKComponentViewAttributeValue CKComponentPanGestureAttribute(CKComponentAction action);
+CKComponentViewAttributeValue CKComponentPanGestureAttribute(const CKTypedComponentAction<UIGestureRecognizer *> &action);
 
 /**
  Returns a view attribute that creates and configures a long press gesture recognizer to send the given CKComponentAction.
-
+ 
  @param action Sent up the responder chain when a long press occurs. Sender is the component that created the view.
-        Context is the gesture recognizer. May be NULL, in which case no action will be sent.
+ Context is the gesture recognizer. May be NULL, in which case no action will be sent.
  */
-CKComponentViewAttributeValue CKComponentLongPressGestureAttribute(CKComponentAction action);
+CKComponentViewAttributeValue CKComponentLongPressGestureAttribute(const CKTypedComponentAction<UIGestureRecognizer *> &action);
```

Comment:
Sadly the C++ templating system is unaware of obj-c's loose typing system, so it requires the exact type in any specializations of the template.